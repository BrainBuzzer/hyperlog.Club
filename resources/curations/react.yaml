name: ReactJS
description: React is a JavaScript library for building user interfaces. It is maintained by Facebook and a community of individual developers and companies.
topics:
  - name: React Basics
    lessons:
      - name: Official ReactJS Tutorial
        link: https://reactjs.org/tutorial/tutorial.html?ref=hyperlog.club
        description: This is official ReactJS tutorial which should work as a good guide to get you up and running with certain aspects of ReactJS.
        difficulty: green
      - name: React Tutorial by Tania Rascia
        link: https://www.taniarascia.com/getting-started-with-react/?ref=hyperlog.club
        description: This tutorial by Tania Rascia gets you from zero to introducing all the concepts in the React ecosystem.
        difficulty: red
  - name: Intermediate React
    lessons:
      - name: Official ReactJS Tutorial
        link: https://reactjs.org/tutorial/tutorial.html?ref=hyperlog.club
        description: This is official ReactJS tutorial which should work as a good guide to get you up and running with certain aspects of ReactJS.
        difficulty: green
      - name: React Tutorial by Tania Rascia
        link: https://www.taniarascia.com/getting-started-with-react/?ref=hyperlog.club
        description: This tutorial by Tania Rascia gets you from zero to introducing all the concepts in the React ecosystem.
        difficulty: green
  - name: Advanced React
    lessons:
      - name: Official ReactJS Tutorial
        link: https://reactjs.org/tutorial/tutorial.html?ref=hyperlog.club
        description: This is official ReactJS tutorial which should work as a good guide to get you up and running with certain aspects of ReactJS.
        difficulty: green
      - name: React Tutorial by Tania Rascia
        link: https://www.taniarascia.com/getting-started-with-react/?ref=hyperlog.club
        description: This tutorial by Tania Rascia gets you from zero to introducing all the concepts in the React ecosystem.
        difficulty: green
resources:
  - name: Awesome React
    link: https://github.com/enaqx/awesome-react?ref=hyperlog.club
    description: A collection of awesome things regarding React Ecosystem.
