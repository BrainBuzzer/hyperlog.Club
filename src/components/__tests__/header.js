import React from "react"
import { render } from "@testing-library/react"

import Header from "../header"

describe("Header", () => {
  it("renders correctly", () => {
    const { asFragment }= render(<Header />)
    expect(asFragment()).toMatchSnapshot()
  })
})
