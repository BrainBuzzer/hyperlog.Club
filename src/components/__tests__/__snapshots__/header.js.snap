// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class=""
    >
      <header
        class="header"
      >
        <div
          class="ant-row"
        >
          <div
            class="ant-col ant-col-6"
          >
            <div
              class="header-logo"
            >
              h
              <span
                class="header-logo-letter-y"
              >
                y
              </span>
              perlog
              <span
                class="header-logo-club"
              >
                .Club
              </span>
            </div>
          </div>
          <div
            class="ant-col ant-col-18"
          >
            <div
              style="float: right; margin-right: 1rem;"
            >
              <ul
                class="ant-menu ant-menu-light ant-menu-root ant-menu-horizontal"
                role="menu"
              >
                <li
                  class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
                  role="menuitem"
                  style="display: none;"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="ant-menu-submenu-title"
                  >
                    <span>
                      ···
                    </span>
                    <i
                      class="ant-menu-submenu-arrow"
                    />
                  </div>
                </li>
                <li
                  class="ant-menu-item ant-menu-item-selected"
                  role="menuitem"
                >
                  <a
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li
                  class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
                  role="menuitem"
                  style="display: none;"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="ant-menu-submenu-title"
                  >
                    <span>
                      ···
                    </span>
                    <i
                      class="ant-menu-submenu-arrow"
                    />
                  </div>
                </li>
                <li
                  class="ant-menu-item"
                  role="menuitem"
                >
                  <a
                    href="/resources"
                  >
                    Resources
                  </a>
                </li>
                <li
                  class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
                  role="menuitem"
                  style="display: none;"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="ant-menu-submenu-title"
                  >
                    <span>
                      ···
                    </span>
                    <i
                      class="ant-menu-submenu-arrow"
                    />
                  </div>
                </li>
                <li
                  aria-disabled="true"
                  class="ant-menu-item join-menu ant-menu-item-disabled"
                  role="menuitem"
                  style="cursor: default;"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-round"
                    type="button"
                  >
                    <a
                      href="/join"
                    >
                      Join Community
                    </a>
                  </button>
                </li>
                <li
                  class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
                  role="menuitem"
                  style="visibility: hidden; position: absolute; display: none;"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="ant-menu-submenu-title"
                  >
                    <span>
                      ···
                    </span>
                    <i
                      class="ant-menu-submenu-arrow"
                    />
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </header>
    </div>
  </div>
</DocumentFragment>
`;
